app/transactions/documents/update_transaction.rb

```diff
@@ -0,0 +1,42 @@
+module Documents
+  class UpdateTransaction
+    include Dry::Transaction
+
+    step :change_filename
+    step :update_file
+
+    def change_filename(params)
+      blob = get_blob(params[:document])
+      new_filename = "#{params[:file_name]}#{blob.filename.extension_with_delimiter}"
+
+      if params[:file_name].blank?
```

Comment:
You can actually make a guard clause for this conditional statement, as if there is no `params[:file_name]`, we don't need to process anything; so no need to `get_blob`, nor contructing the `new_filename`:
```ruby
def change_filename(params)
  return Success(params) if params[:file_name].blank?

  blob = get_blob(params[:document])
...
```